evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -59,6 +59,52 @@ def transform(self, X, y=None):
             y = infer_feature_types(y)
         return X, None
 
+    def _convert_dictionary(self, sampling_dict, y):
+        """Converts the provided sampling dictionary from a dictionary of ratios to a dictionary of number of samples.
+        Expects the provided dictionary keys to be the target values y, and the associated values to be the min:max ratios.
+        Converts and returns a dictionary with the same keys, but changes the values to be the number of samples rather than ratio.
+
+        Arguments:
+            sampling_dict (dict): The input sampling dictionary passed in from user
+            y (ww.DataColumn): The target values
+
+        Returns:
+            dict: A dictionary with target values as keys and the number of samples as values
+        """
+        # check that the lengths of the dict and y are equal
+        y_unique = y.unique()
+        if len(sampling_dict) != len(y_unique):
```

Comment:
I think the below check will catch this case?

Reply:
Yeah, I wanted to include both since they are checking for slightly different things. Do you think this first case is unnecessary?

Reply:
I see! Can we make the error message more detailed for the first case then?, e.g. sampling_dict contains different number of target values than are in the data or something like that?

Reply:
Suggesting because if we keep the first case, I feel the error message should be more actionable than "Lengths are different"

Reply:
Done!

Reply:
🚢